test/sai_test/config/route_configer.py

```diff
@@ -494,212 +309,84 @@ def create_neighbor_by_rif(self, nexthop_device: Device, rif, virtual_router=Non
-
-    def create_router_interface_by_lag(self, lag: Lag, virtual_router=None):
-        """
-        Create lag intreface.
-        It will check if the lag already created a route interface
-
-        Set lag attribute lag_id
-
-        Attrs:
-            lag: Lag object that this lag interface mapping
-            virtual_router_id: virtual route id, if not defined, will use default route
-            virtual_router_id: virtual route id, if not defined, will use default route
-
-        return rif
-        """
-        if not lag.rif:
-            vr_id = self.choice_virtual_route(virtual_router)
-            rif = sai_thrift_create_router_interface(self.client,
-                                                     virtual_router_id=vr_id,
-                                                     type=SAI_ROUTER_INTERFACE_TYPE_PORT,
-                                                     port_id=lag.lag_id)
-            lag.rif = rif
-            self.test_obj.assertEqual(
-                self.test_obj.status(), SAI_STATUS_SUCCESS)
-        return lag.rif
-
-    def create_router_interface_by_port_idx(self, port_idx, virtual_router=None):
+    def create_router_interface_by_port_idx(self, port_idx, virtual_router=None, reuse=True, is_bridge=False):
         """
         Create route interface by port index for a port.
-        It will check if the port already created a route interface
-
-        Set dut attribute port_rif_list
+        It will check if the port already created on a route interface. If 'reuse',
+        it will return the last one created for this port, if not 'reuse', it will create a new one,
+        and store it with this port and dut object.
 
         Attrs:
             port_idx: port index
             virtual_router_id: virtual route id, if not defined, will use default route
-
+            reuse: reuse the existing rif which is binding to the port
+            is_bridge: is a bridge port, only used for port
         return: route interface
         """
-        if not self.test_obj.dut.port_rif_list[port_idx]:
-            vr_id = self.choice_virtual_route(virtual_router)
-            port_id = self.test_obj.dut.port_list[port_idx]
-            rif_id = sai_thrift_create_router_interface(
-                self.client, virtual_router_id=vr_id, type=SAI_ROUTER_INTERFACE_TYPE_PORT, port_id=port_id)
-            self.test_obj.dut.port_rif_list[port_idx] = rif_id
-            self.test_obj.assertEqual(
-                self.test_obj.status(), SAI_STATUS_SUCCESS)
-
-        return self.test_obj.dut.port_rif_list[port_idx]
+        net_intf = self.test_obj.dut.port_obj_list[port_idx]
+        return self.create_router_interface(netItf=net_intf, virtual_router=virtual_router, reuse=reuse, is_bridge=is_bridge)
 
-    def create_router_interface_by_bridge_port_idx(self, bridge_port_idx, virtual_router=None):
+    def create_router_interface(self, netItf: route_item, virtual_router=None, reuse=True, is_bridge=False):
```

Comment:
Why suddenly switch to camelCase to name this netItf variable?

Reply:
fixed